sap-actions/src/automation/sap/utils/credentialAttributeMapper.ts

```diff
@@ -0,0 +1,42 @@
+interface DecryptedCredential {
+    id: string;
+    name: string;
+    order: number;
+    template: string;
+    attributes: {
+        id: string;
+        name: string;
+        value: string;
+    }[];
+}
+
+
+/**
+ * Takes array of decrypted credentials and returns mapped credential attributes in expected type.
+ *
+ * @param credentials Single element array of decrypted credentials matched for specific action
+ * @returns Mapped credential attributes to single object with expected type.
+ */
+export const credentialAttributesMapper = <R = Record<string, any>>(
+    credentials: DecryptedCredential[]
+): R | undefined => {
+    if (!credentials || !credentials.length) return;
+    if (credentials.length > 1) {
+        throw new Error('More than one credential per action is not allowed');
+    }
+
+    const { attributes } = credentials[0];
+    if (!attributes.length) {
+        throw new Error('Credential must have at least one attribute');
+    }
+
+    const mappedCredential = attributes
+    .reduce((acc, { name, value }) => {
+        acc[name] = value;
+        return acc;
+    }, {} as R);
+
+    return Object.values(mappedCredential).length
+        ? mappedCredential
+        : undefined;
```

Comment:
can you also fix this part? I think it would be better to just return `mappedCredential` (remember to adjust return type to return `R | {}`)